src/extensions/surveys.tsx

```diff
@@ -320,40 +315,66 @@ export function Questions({
             setCurrentQuestion(idx + 1)
         }
     }
-    const isMultipleQuestion = survey.questions.length > 1
+
+    const questionToDisplay = readOnly ? previewQuestionIndex : currentQuestion
```

Comment:
A bit annoying that in the main code we have to have special handling for previews, I expect this will become a lot more annoying when you need actual functionality with the branching logic for example.

Reply:
💯 , I want to clean this up